### PR TITLE
Add scsi hostdev and controller cases

### PR DESCRIPTION
**Pull request by chunfuwen**
Add scsi hostdev and controller cases
xx-98979,xx-54603,xx-103724,xx-202693,xx-186120,xx-172931,xx-134267

### CODE REVIEW

**Comment by chunfuwen** [comment]
PASS 1-type_specific.io-github-autotest-libvirt.scsi_device_manipulate.coldplug.scsi_hostdev.positive.alias
PASS 2-type_specific.io-github-autotest-libvirt.scsi_device_manipulate.coldplug.scsi_hostdev.positive.boot_order
PASS 4-type_specific.io-github-autotest-libvirt.scsi_device_manipulate.coldplug.scsi_hostdev.negative.max_sectors_controller
PASS 5-type_specific.io-github-autotest-libvirt.scsi_device_manipulate.hotplug.scsi_hostdev.positive.same_hostdev_address
PASS 6-type_specific.io-github-autotest-libvirt.scsi_device_manipulate.hotplug.scsi_hostdev.positive.tap_library
PASS 7-type_specific.io-github-autotest-libvirt.scsi_device_manipulate.hotplug.scsi_hostdev.positive.unplug_scsi_controller

**Review by dzhengfy — COMMENTED** [review]
Others LGTM

File: libvirt/tests/src/scsi/scsi_device.py [hunk 1]
```diff
@@ -0,0 +1,510 @@
...
+    # Back up xml file
+    if vm.is_alive():
+        vm.destroy(gracefully=False)
+
+    coldplug = "yes" == params.get("coldplug")
+    define_error = "yes" == params.get("define_error", "no")
+    xml_backup = vm_xml.VMXML.new_from_inactive_dumpxml(vm_name)
+
+    plug_mode = params.get("plug_mode")
+    test_operation = params.get("test_operation")
```
Comment (smitterl) [inline libvirt/tests/src/scsi/scsi_device.py:455]: Is this necessary? The tests are below `scsi/scsi_device.(py|cfg)` and the only variant is `scsi_hostdev`, that is, it's value is constant.

File: libvirt/tests/cfg/scsi/scsi_device.cfg [hunk 2]
```diff
@@ -0,0 +1,45 @@
+- scsi_device_manipulate:
+    type = scsi_device
+    status_error = "no"
+    start_vm = "no"
+    variants test_operation:
```
[the remark this answers is not in the document]
Comment (chunfuwen) [reply libvirt/tests/cfg/scsi/scsi_device.cfg]: updated

File: libvirt/tests/src/scsi/scsi_device.py (hunk 1 above)
Comment (chunfuwen) [reply libvirt/tests/src/scsi/scsi_device.py]: @smitterl 
thanks for pointing this out. it aims to address possible future variant extension. For example, scsi_type may have multiple value, not only currently available : scsi_hostdev

**Comment by chunfuwen** [comment]
@dzhengfy 
@smitterl 
comments are updated, please double check them

**Review by dzhengfy — APPROVED** [review]
LGTM

**Review by smitterl — APPROVED** [review]
LGTM

**Pull request merged**